src/routes/Trip/Items.tsx

```diff
@@ -49,6 +51,15 @@ const Items = ({
 
   const isListPresent = currentTrip && listsById[listId];
 
+  let listOfItemIds = [];
+
+  const queryHandler = (value: any) => {
+    //Value from the FilterItemForm
+
+    setItemQuery(value);
+  };
+
+  const itemFilterIds: any[] = [];
```

Comment:
A few suggestions here: 

1. maybe we move `listOfItemIds` and `itemFilterIds` a bit closer to the filtering process?

2. Maybe we rename the empty array `itemFilterIds` to `filteredItemIds` ?
and 

3. I'm thinking we can directly assign all the list's ids to `listOfItemIds`, then the final mapping (for checked and unchecked) will only happen over `filteredItemIds`.  

```
  if (isListPresent) {
    const listOfItemIds = listsById[currentTrip.listId].itemIds;
    let filteredItemIds = [];
```

This way, if the query is empty, the `filteredItemIds` will be all of the `listOfItemIds`

```
    if (itemQuery === '') {
      //Check if form field has any value
      filteredItemIds = listOfItemIds;

    } else {
      filteredItemIds = listOfItemIds.filter((itemId:any)=>{
        return items[itemId].label.toLowerCase().includes(itemQuery.toLocaleLowerCase());
      });
    }
```